lib/ansible/modules/cloud/amazon/cloudformation_facts.py

```diff
@@ -184,18 +192,21 @@ def describe_stacks(self, stack_name=None):
                 return response
             self.module.fail_json(msg="Error describing stack(s) - an empty response was returned")
         except Exception as e:
-            self.module.fail_json(msg="Error describing stack(s) - " + str(e), exception=traceback.format_exc())
+            if 'does not exist' in e.message:
```

Comment:
Shouldn't we use e.response['Error']['Code']?

Reply:
Good catch, thanks. I've used e.response['Error']['Message'] since the error code is the vague 'ValidationError'.

Reply:
For some reason my pull request isn't updating. I pushed a commit https://github.com/s-hertel/ansible/commit/a75d62b9f6feeec363ffd470cfbad38901e3b8c9 to address your review. Might close this and open a new PR if it doesn't sync with my branch... Weird.